### DISCUSSION TITLE
Updating the build selected for submission to AppStore without starting a review

### DISCUSSION THREAD

**Discussion by shalamowww** [fastlane/fastlane#17506] [Q&A] [answered]
I have a couple of copies of a main app in AppStore, so I can deliver the same app under different bundle IDs with different feature sets.
The copies don't get their version incremented and stay at version 1.0. 
When a new build is uploaded for a copy, I want it to be selected for the "submission" (I have no intention of submitting any of these apps, they are internal) so that the icon on AppStore connect changes (I use badge to add info to the icon).
Is there a way to do so?

I use this code:
```
upload_to_testflight(changelog: ENV["DESCRIPTION"])

if options[:type] != "release"
	deliver(
    		build_number: build_number, 
    		submit_for_review: false,
    		automatic_release: false,
    		force: true,
    		skip_metadata: true,
    		skip_screenshots: true,
    		skip_binary_upload: true
  	)
end
```
It completes with no errors, but doesn't do anything. 
Could you confirm if it is even possible and point me to a corresponding part of the docs please?
If not, I'd like to try making a PR for such a feature and would welcome any tips on where to look for a piece of code to modify.

**Comment by rogerluan** [comment]
I do something somewhat similar, but I'm not sure if the build is selected 😅 - I know the metadata and all the other info is. Later then I call submit_for_review=true and then I'm sure the build is selected, but I can't vouch for the first case.
The difference I see in the commands is that I set the `app_version` - and not the build number, but I see that you need the build number. Even if it's always `1.0.0`, maybe you could try sending the `app_version` too? 🤔 

If that doesn't work, I'm not sure this will be possible 🤷 Although it sounds like it should be, it may be a bug that it doesn't do anything. 
Also, I'm assuming you double checked the build numbers and all, right? 😁 

Good luck!

**Comment by rogerluan** [comment, marked as answer]
Forget what I said above ☝️ I checked the code and actually it won't select the build, unfortunately. See https://github.com/fastlane/fastlane/discussions/17556 for more info 😬 

You can workaround this limitation by calling e.g. `version.select_build(build_id: build_id)` manually from Spaceship, in ruby. This can be done through your Fastfile if you're using one in ruby, or via a standalone ruby script.
